addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_38/A5_38_08_Dimming.java

```diff
@@ -44,31 +53,78 @@ public A5_38_08_Dimming(ERP1Message packet) {
     @Override
     protected void convertFromCommandImpl(String channelId, String channelTypeId, Command outputCommand,
             State currentState, Configuration config) {
+        switch (channelId) {
+            case CHANNEL_DIMMER:
+                byte dimmValue;
+
+                if (outputCommand instanceof DecimalType) {
+                    dimmValue = ((DecimalType) outputCommand).byteValue();
+                } else if (outputCommand instanceof OnOffType) {
+                    dimmValue = ((OnOffType) outputCommand == OnOffType.ON) ? Switch100Percent : Zero;
+                } else if (outputCommand instanceof IncreaseDecreaseType) {
+                    dimmValue = ((IncreaseDecreaseType) outputCommand == IncreaseDecreaseType.INCREASE)
+                            ? Switch100Percent
+                            : Zero;
+                } else if (outputCommand instanceof UpDownType) {
+                    dimmValue = ((UpDownType) outputCommand == UpDownType.UP) ? Switch100Percent : Zero;
+                } else {
+                    throw new IllegalArgumentException(outputCommand.toFullString() + " is no valid dimming command.");
+                }
+
+                EnOceanChannelDimmerConfig c = config.as(EnOceanChannelDimmerConfig.class);
+
+                boolean eltakoDimmer = (c.eltakoDimmer == null) ? true : c.eltakoDimmer;
```

Comment:
If you have default values assigned, you can even use "boolean" instead of the wrapped "Boolean" in the configuration class, and remove those local variables. Also true for `rampingTime` further down.

Reply:
Ok, changed here: #5055